lib/thor.rb

```diff
@@ -327,6 +327,8 @@ def subcommand_classes
     end
 
     def subcommand(subcommand, subcommand_class)
+      return subcommand_classes[subcommand.to_s] if subcommands.include? subcommand.to_s
```

Comment:
Do we need to return the command class? It seems we only need to not register the command again. But what happens if people wants to override the subcommand to have different code? Let's say a library trying to enhance an already existing class. It seems that would break that behavior.